safe/gui/tools/wizard/step_kw45_extrakeywords.py

```diff
@@ -104,116 +103,19 @@ def get_next_step(self):
-
-        :param indx: The new index.
-        :type indx: int or str
-        """
-        if isinstance(indx, int) and indx > -1:
-            self.extra_keyword_changed(self.extra_keywords_widgets[1])
-
-    # noinspection PyPep8Naming
-    def on_cboExtraKeyword3_currentIndexChanged(self, indx):
-        """This is an automatic Qt slot executed when the
-           3rd extra keyword combobox selection changes.
-
-        :param indx: The new index.
-        :type indx: int or str
-        """
-        if isinstance(indx, int) and indx > -1:
-            self.extra_keyword_changed(self.extra_keywords_widgets[2])
-
-    # noinspection PyPep8Naming
-    def on_cboExtraKeyword4_currentIndexChanged(self, indx):
-        """This is an automatic Qt slot executed when the
-           4th extra keyword combobox selection changes.
-
-        :param indx: The new index.
-        :type indx: int or str
-        """
-        if isinstance(indx, int) and indx > -1:
-            self.extra_keyword_changed(self.extra_keywords_widgets[3])
-
-    # noinspection PyPep8Naming
-    def on_cboExtraKeyword5_currentIndexChanged(self, indx):
-        """This is an automatic Qt slot executed when the
-           5th extra keyword combobox selection changes.
-
-        :param indx: The new index.
-        :type indx: int or str
-        """
-        if isinstance(indx, int) and indx > -1:
-            self.extra_keyword_changed(self.extra_keywords_widgets[4])
-
-    # noinspection PyPep8Naming
-    def on_cboExtraKeyword6_currentIndexChanged(self, indx):
-        """This is an automatic Qt slot executed when the
-           6th extra keyword combobox selection changes.
-
-        :param indx: The new index.
-        :type indx: int or str
-        """
-        if isinstance(indx, int) and indx > -1:
-            self.extra_keyword_changed(self.extra_keywords_widgets[5])
-
-    # noinspection PyPep8Naming
-    def on_cboExtraKeyword7_currentIndexChanged(self, indx):
-        """This is an automatic Qt slot executed when the
-           7th extra keyword combobox selection changes.
-
-        :param indx: The new index.
-        :type indx: int or str
-        """
-        if isinstance(indx, int) and indx > -1:
-            self.extra_keyword_changed(self.extra_keywords_widgets[6])
-
-    # noinspection PyPep8Naming
-    def on_cboExtraKeyword8_currentIndexChanged(self, indx):
-        """This is an automatic Qt slot executed when the
-           8th extra keyword combobox selection changes.
-
-        :param indx: The new index.
-        :type indx: int or str
-        """
-        if isinstance(indx, int) and indx > -1:
-            self.extra_keyword_changed(self.extra_keywords_widgets[7])
+        if (self.parent.get_layer_geometry_key() ==
+                layer_geometry_raster['key']):
+            return []
+        # Get hazard or exposure value
+        subcategory = self.parent.step_kw_subcategory.selected_subcategory()
+        return get_fields(
```

Comment:
Can you make a temporary var ?
It's difficult to debug when there is a `return` and a function at the same time.